src/typedefs.rs

```diff
@@ -85,6 +85,59 @@ macro_rules! impl_normal_debug_trait (($name:ident) => (
     }
 ));
 
+/// Macro that implements the `serde::{Serialize, Deserialize}` traits.
+#[cfg(feature = "serde")]
+macro_rules! impl_serde_traits (($name:ident, $bytes_function:ident) => (
+    /// This type tries to serialize as a `&[u8]` would. Note that the serialized
+    /// type likely does not have the same protections that orion provides, such
+    /// as constant-time operations. A good rule of thumb is to only serialize
+    /// these types for storage. Don't operate on the serialized types.
+    impl serde::Serialize for $name {
+        fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+        where
+            S: serde::ser::Serializer,
+        {
+            let bytes: &[u8] = self.$bytes_function();
+            bytes.serialize(serializer)
+        }
+    }
+
+    /// This type tries to deserialize as a `Vec<u8>` would. If it succeeds, the digest
+    /// will be built using `Self::from_slice`.
+    ///
+    /// Note that **this allocates** once to store the referenced bytes on the heap.
+    impl<'de> serde::Deserialize<'de> for $name {
+        fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+        where
+            D: serde::de::Deserializer<'de>,
+        {
+            let bytes = Vec::<u8>::deserialize(deserializer)?;
+            std::convert::TryFrom::try_from(bytes.as_slice()).map_err(serde::de::Error::custom)
+        }
+    }
+));
+
+#[cfg(test)]
+#[cfg(feature = "serde")]
+macro_rules! test_serde_impls (($name:ident, $gen_length:expr) => (
+    #[test]
+    fn test_serde_serialized_equivalence_to_bytes_fn() {
```

Comment:
We have to have at least one test in the `Tests` workflow compile with the currently-missing `serde` feature, such that these tests are run in CI.

Reply:
I think `serde` is the first non-default optional feature added to Orion. Maybe the right way forward is to have an optional feature called `all` or `extra` that will explicitly enable all extra features, including `serde` in this case.

The counterpoint to this is that we don't have any other non-default optional features at the moment, and this could be an example of premature generalization. I could go either way.

For now, for testing, I'll add a new test that's just like normal, except with `serde` enabled. In the future, if we want to add more optional features, maybe we replace that test with something like `cargo test --features all`.